tests/late_joiners.py

```diff
@@ -53,46 +53,46 @@ def find_primary(network):
     return term_info
 
 
-def timeout(node, suspend, election_timeout):
-    if suspend:
-        # We want to suspend the nodes' process so we need to initiate a new timer to wake it up eventually
-        node.suspend()
-        next_timeout = random.uniform(2 * election_timeout, 3 * election_timeout)
-        LOG.info(f"New timer set for node {node.node_id} is {next_timeout} seconds")
-        t = Timer(next_timeout, timeout, args=[node, False, 0])
-        t.start()
-    else:
-        node.resume()
-
-
-def assert_node_up_to_date(check, node, final_msg, final_msg_id):
+def assert_node_up_to_date(check, node, final_msg, final_msg_id, timeout=5):
     with node.user_client() as c:
-        for x in range(0, 5):
+        while timeout > 0:
```

Comment:
We'll switch to a library but for now let's do timeouts consistently with the code now that Eddy has updated it and use elapsed time.